libs/npm/core/src/utils/fromQueryString.ts

```diff
@@ -1,22 +1,39 @@
+export interface IQueryOptions {
+  /** An array of query parameters that should be converted to arrays. */
+  arrays?: string[];
+  /** An array of query parameters that should be converted to numbers. */
+  numbers?: string[];
+}
+
 /**
  * Converts a query string to an object.
  * @param query The query string to convert to an object.
+ * @param options Query options to control the output.
  * @returns A new instance of an object.
  */
-export const fromQueryString = (query?: string): any => {
+export const fromQueryString = (query?: string, options?: IQueryOptions): any => {
```

Comment:
We can now properly handle arrays that only have a single value in the query parameters.